example/cumulus-tf/vpc.tf

```diff
@@ -5,10 +5,12 @@ data "aws_vpc" "application_vpc" {
   }
 }
 
-data "aws_subnet_ids" "subnet_ids" {
+data "aws_subnets" "subnet_ids" {
   count = length(var.lambda_subnet_ids) == 0 ? 1 : 0
-  vpc_id = var.vpc_id != null ? var.vpc_id : data.aws_vpc.application_vpc[0].id
-
+  filter {
```

Comment:
What does `filter` do in this context? From what I've seen it's supposed to "narrow down" results but here we have a block that just assigns values...?

Reply:
So - the object type has changed here from aws_subnet_ids to aws_subnets (https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet.html).

Unlike aws_subnet_ids (which uses vpc_id to return , the aws_subnets data source returns *all* subnets for an account - this is not what we want as we *only* only subnets in the VPC Core is set to work in.    Bear in mind we're using this only for defaults in the example project where var.lambda_subnet_ids is *not* set (it will instead grab a single subnet at random-ish from a vpc subnet *or* the application_vpc if they also do not have a vpc configured. 

So, to answer the question, in this context, filter is filtering out a single aws subnet that's in the VPC Core is in, or the first application_vpc if for some reason that's not configured.